aspnetcore/host-and-deploy/scaling-aspnet-apps/scaling-aspnet-apps.md

```diff
@@ -0,0 +1,320 @@
+
+## Set up the sample project
+
+You can use the GitHub Explorer sample app to follow along with this tutorial. Clone the app from GitHub using the following command:
+
+```dotnetcli
+git clone "https://github.com/dotnet/AspNetCore.Docs.Samples.git"
+```
+
+Navigate to the `/tutorials/scalable-razor-apps/start` folder and double select the `ScalableRazor.csproj` file to open the project in Visual Studio.
+
+The sample app uses a search form to browse GitHub repositories by name. The form relies on the built-in ASP.NET Core data protection services to handle anti-forgery concerns. By default, when the app scales horizontally on Container Apps, the data protection service throws an exception. You'll explore and solve this issue in the steps ahead.
+
+#### Test the app
+
+1. Open the project in Visual Studio by double clicking on the `GitHubBrowser.sln` solution file in the project folder.
+1. Launch the app by clicking the run button at the top of Visual Studio. The project includes a Docker file, which means you can select the arrow next to the run button to start the app using either a Docker Desktop setup or the standard ASP.NET Core local web server.
+
+When the app launches in the browser, you can use the search form to browse for GitHub repositories by name.
+
+:::image type="content" source="./media/scaling-app-screenshot.png" alt-text="A screenshot showing the GitHub Explorer app.":::
+
+## Deploy the app to Azure Container Apps
+
+You'll use Visual Studio to deploy the app to Azure Container Apps. Container apps provide a managed service designed to simplify hosting containerized apps and microservices.
+
+1. Inside the Visual Studio solution explorer, right click on the top level project node and select **Publish**.
+1. In the publishing dialog, select **Azure** as the deployment target, and then choose **Next**.
+1. For the specific target, select **Azure Container Apps (Linux)**, and then choose **Next**.
+1. You'll need to create a new container app to deploy to. Select the green **+** icon to open a new dialog and enter the following values:
+
+    :::image type="content" source="./media/scaling-deploy-visual-studio-small.png" lightbox="./media/scaling-deploy-visual-studio.png" alt-text="A screenshot showing Visual Studio deployment.":::
+
+    * **Container app name**: Leave the default value or enter a name of your choosing.
+    * **Subscription name**: Select the subscription you'd like to deploy to.
+    * **Resource group**: Select **New** and create a new resource group called *msdocs-scalable-razor*.
+    * **Container apps environment**: Select **New** to open the container apps environment dialog and enter the following values:
+        * **Environment name**: Leave the default value or enter a name of your choosing.
+        * **Location**: Select a location near you.
```

Comment:
Maybe we should make a statement up front about location. Off the top of my head, something like
Many of the resources created in the app require a location. For this app, location isn't important. A real app should select a location closest to the clients. You may want to select a location near you.

Some of the resources have locations that others don't, so my suggestion to use the previous location won't work.

Reply:
Yes, I was hesitant to pick a specific location and then always refer to that same one since I know Container Apps and some other services aren't as widely available across regions as others. I'll try to add some kind of note about it.